src/store/root.reducer.ts

```diff
@@ -1,21 +1,12 @@
-const initiatState = {
-  data: {},
-  petsToCompare: [],
-  search: {},
-};
+import { createReducer } from 'typesafe-actions';
 
-export const rootReducer = (state: object = initiatState, action: any) => {
-  switch (action.type) {
-    case 'PUT_PETS':
-      return {
-        ...state,
-        petsToCompare: action.payload.pets,
-      };
-    case 'SEARCH':
-      return {
-        search: action.payload,
-      };
-  }
+type State = {} | [];
+type Action = { type: 'FILTER'; payload: {} } | { type: 'PUT_PETS'; payload: any };
```

Comment:
Do not write action types on your own. You should be able to infer them automatically.

See the documentation: https://github.com/piotrwitek/typesafe-actions#alternative-usage-with-regular-switch-reducer